### PR TITLE
Clean as much as possible location label for exchange events

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/docs/blob/main/usage-guides/index.md) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/db/upgrades/v52_v53.py [hunk 1]
```diff
@@ -78,6 +79,67 @@ def _add_gate_location(write_cursor: 'DBCursor') -> None:
...
+            Location.KUCOIN,
+            Location.NEXO,
+            Location.OKX,
+            Location.POLONIEX,
+            Location.SHAPESHIFT,
+            Location.UPHOLD,
+            Location.WOO,
+        ))
+        placeholders = ','.join('?' * len(exchange_locations))
+        write_cursor.execute(
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v52_v53.py:126]: The fill UPDATE has a Halloween problem: its COUNT(DISTINCT location_label) <= 1 check reads history_events while the same statement mutates it. For a location with 1 credential and one existing label that differs from the key name (e.g. a renamed API key), the first
  NULL row fills, the distinct count flips to 2, and the rest get skipped — one row filled, the rest left NULL.                                                 

  Can you freeze the eligibility into a temp table before writing, then have the UPDATE read from that snapshot?                                                                                                                                                                 
```
  write_cursor.execute(
      f'CREATE TEMP TABLE _label_fill AS '
      f'SELECT uc.location AS location, uc.name AS name FROM user_credentials uc '
      f'WHERE uc.location IN ({placeholders}) GROUP BY uc.location HAVING COUNT(*)=1 AND ('
      'SELECT COUNT(DISTINCT he.location_label) FROM history_events he '
      'WHERE he.location=uc.location AND he.location_label IS NOT NULL)<=1',
      exchange_locations,
  )
  write_cursor.execute(
      'UPDATE history_events SET location_label='
      '(SELECT name FROM _label_fill WHERE location=history_events.location) '
      'WHERE location IN (SELECT location FROM _label_fill) AND location_label IS NULL',
  )
  write_cursor.execute('DROP TABLE _label_fill')
```

  Eligibility is now computed once against a snapshot taken before any write, so filling one row can't change whether the others qualify — all NULL rows fill regardless of order (also drops the per-row subqueries). Can you also add a test case with ≥2 NULL rows where the
  existing label differs from the key name? That's the gap that lets the current bug through.

File: rotkehlchen/db/upgrades/v52_v53.py [hunk 2]
```diff
@@ -78,6 +79,67 @@ def _add_gate_location(write_cursor: 'DBCursor') -> None:
...
+            Location.ICONOMI,
+            Location.INDEPENDENTRESERVE,
+            Location.KRAKEN,
+            Location.KUCOIN,
+            Location.NEXO,
+            Location.OKX,
+            Location.POLONIEX,
+            Location.SHAPESHIFT,
+            Location.UPHOLD,
+            Location.WOO,
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v52_v53.py:123]: GATE is missing here. I guess since we add it in devleop it's not needed bu at least mention it as a comment

File: rotkehlchen/db/upgrades/v52_v53.py [hunk 3]
```diff
@@ -78,6 +79,67 @@ def _add_gate_location(write_cursor: 'DBCursor') -> None:
...
+            'SELECT name FROM user_credentials WHERE location=history_events.location) '
+            f'WHERE location IN ({placeholders}) AND location_label IS NULL AND '
+            '(SELECT COUNT(*) FROM user_credentials '
+            'WHERE location=history_events.location)=1 AND '
+            '(SELECT COUNT(DISTINCT existing_events.location_label) FROM history_events AS '
+            'existing_events WHERE existing_events.location=history_events.location AND '
+            'existing_events.location_label IS NOT NULL)<=1',
+            exchange_locations,
+        )
+        write_cursor.execute(
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v52_v53.py:136]: this will nullify location label for any exchange users may have added visa CSV, any dead exchange like say FTX, Bittrex etc.

Is this intended? And if so why? Because we only want to have the label set if an api key is set? State it if so

File: rotkehlchen/db/upgrades/v52_v53.py (hunk 2 above)
Comment (yabirgb) [reply rotkehlchen/db/upgrades/v52_v53.py:123]: yes, since it is in develop I removed it intentionally. added the comment

File: rotkehlchen/db/upgrades/v52_v53.py (hunk 3 above)
Comment (yabirgb) [reply rotkehlchen/db/upgrades/v52_v53.py:136]: The CSV imports were not intended. I've modified the PR, now only fills missing labels when it is safe to infer a single API key label, and leaves existing non null labels untouched.

File: rotkehlchen/db/upgrades/v52_v53.py [hunk 4]
```diff
@@ -78,6 +79,66 @@ def _add_gate_location(write_cursor: 'DBCursor') -> None:
...
+            Location.NEXO,
+            Location.OKX,
+            Location.POLONIEX,
+            Location.SHAPESHIFT,
+            Location.UPHOLD,
+            Location.WOO,
+        ))
+        placeholders = ','.join('?' * len(exchange_locations))
+        write_cursor.execute(
+            f'CREATE TEMP TABLE _exchange_label_fill AS SELECT user_credentials.location AS '
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v52_v53.py]:
```suggestion
            f'CREATE TEMP TABLE IF NOT EXISTS _exchange_label_fill AS SELECT user_credentials.location AS '
```

File: rotkehlchen/db/upgrades/v52_v53.py [hunk 5]
```diff
@@ -78,6 +79,66 @@ def _add_gate_location(write_cursor: 'DBCursor') -> None:
...
+            'FROM history_events WHERE history_events.location=user_credentials.location AND '
+            'history_events.location_label IS NOT NULL)<=1',
+            exchange_locations,
+        )
+        write_cursor.execute(
+            'UPDATE history_events SET location_label=(SELECT name FROM _exchange_label_fill '
+            'WHERE location=history_events.location) WHERE location IN (SELECT location FROM '
+            '_exchange_label_fill) AND location_label IS NULL',
+        )
+        write_cursor.execute('DROP TABLE _exchange_label_fill')
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v52_v53.py]:
```suggestion
        write_cursor.execute('DROP TABLE IF EXISTS _exchange_label_fill')
```

**Review by LefterisJP — COMMENTED** [review]
a bit more defensive nit

**Review by LefterisJP — APPROVED** [review]
LGTM
